addons/mail/static/src/js/attachment_box.js

```diff
@@ -64,6 +67,68 @@ var AttachmentBox = Widget.extend({
         this.currentResModel = record.model;
     },
 
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Calls controller to upload file, separate method is created
+     * to use it in tests to patch it.
+     *
+     * @param {string} controllerUrl
+     * @param {FormData} formData
+     */
+    _callUploadAttachment: function (controllerUrl, formData) {
+        return $.ajax({
```

Comment:
prefer native solution over `$.ajax`. You may use `window.fetch()`, and it has a very nice API by the way!